tests/dragonfly/conftest.py

```diff
@@ -229,42 +230,89 @@ def memcached_connection(df_server: DflyInstance):
 
 
 @pytest.fixture(scope="session")
-def gen_tls_cert(df_factory: DflyInstanceFactory):
-    tls_server_key_file_name = "df-key.pem"
-    tls_server_cert_file_name = "df-cert.pem"
+def gen_ca_cert(df_factory: DflyInstanceFactory):
     dfly_path = df_factory.dfly_path
     # We first need to generate the tls certificates to be used by the server
 
-    # Step 1
     # Generate CA (certificate authority) key and self-signed certificate
     # In production, CA should be generated by a third party authority
     # Expires in one day and is not encrtypted (-nodes)
     # X.509 format for the key
     ca_key = dfly_path + "ca-key.pem"
     ca_cert = dfly_path + "ca-cert.pem"
-    step1 = rf'openssl req -x509 -newkey rsa:4096 -days 1 -nodes -keyout {ca_key} -out {ca_cert} -subj "/C=GR/ST=SKG/L=Thessaloniki/O=KK/OU=AcmeStudios/CN=Gr/emailAddress=acme@gmail.com"'
-    subprocess.run(step1, shell=True)
+    step = rf'openssl req -x509 -newkey rsa:4096 -days 1 -nodes -keyout {ca_key} -out {ca_cert} -subj "/C=GR/ST=SKG/L=Thessaloniki/O=KK/OU=AcmeStudios/CN=Gr/emailAddress=acme@gmail.com"'
+    subprocess.run(step, shell=True)
+
+    return {"ca_key": ca_key, "ca_cert": ca_cert}
+
 
-    # Step 2
+@pytest.fixture(scope="session")
+def gen_server_cert(df_factory: DflyInstanceFactory, gen_ca_cert):
     # Generate Dragonfly's private key and certificate signing request (CSR)
+    dfly_path = df_factory.dfly_path
+    tls_server_key_file_name = "df-key.pem"
+    tls_server_cert_file_name = "df-cert.pem"
     tls_server_key = dfly_path + tls_server_key_file_name
     tls_server_req = dfly_path + "df-req.pem"
-    step2 = rf'openssl req -newkey rsa:4096 -nodes -keyout {tls_server_key} -out {tls_server_req} -subj "/C=GR/ST=SKG/L=Thessaloniki/O=KK/OU=Comp/CN=Gr/emailAddress=does_not_exist@gmail.com"'
-    subprocess.run(step2, shell=True)
+    step1 = rf'openssl req -newkey rsa:4096 -nodes -keyout {tls_server_key} -out {tls_server_req} -subj "/C=GR/ST=SKG/L=Thessaloniki/O=KK/OU=Comp/CN=Gr/emailAddress=does_not_exist@gmail.com"'
+    subprocess.run(step1, shell=True)
 
-    # Step 3
     # Use CA's private key to sign dragonfly's CSR and get back the signed certificate
+    ca_key = gen_ca_cert["ca_key"]
+    ca_cert = gen_ca_cert["ca_cert"]
     tls_server_cert = dfly_path + tls_server_cert_file_name
-    step3 = fr'openssl x509 -req -in {tls_server_req} -days 1 -CA {ca_cert} -CAkey {ca_key} -CAcreateserial -out {tls_server_cert}'
-    subprocess.run(step3, shell=True)
-    return tls_server_key_file_name, tls_server_cert_file_name
+    step2 = fr'openssl x509 -req -in {tls_server_req} -days 1 -CA {ca_cert} -CAkey {ca_key} -CAcreateserial -out {tls_server_cert}'
+    subprocess.run(step2, shell=True)
+
+    return {"server_key": tls_server_key, "server_cert": tls_server_cert}
+
+
+@pytest.fixture(scope="session")
+def gen_client_cert(df_factory: DflyInstanceFactory, gen_ca_cert):
```

Comment:
Is there a real difference between gen_client_cert and gen_server_cert functions?or is it just the file names ? 
If they are the same I think you should create a function gen_cert which you will pass the filenames